### ISSUE TITLE
Missing Marketing Icon - iOS Apps must include a 1024x1024px Marketing Icon in PNG format

### ISSUE THREAD

**Issue by stalinkay** [ionic-team/ionic-cli#2742]
Kindly include a `1024x1024px` Marketing Icon when running `ionic cordova resources`, it is a new requirement for apps built with Xcode 9. See letter below:

```
Dear developer,

We have discovered one or more issues with your recent delivery for "NWR – Namibia Wildlife Resorts". Your delivery was successful, but you may wish to correct the following issues in your next delivery:

Missing Marketing Icon - iOS Apps must include a 1024x1024px Marketing Icon in PNG format. Apps that do not include the Marketing Icon cannot be submitted for App Review or Beta App Review.

After you’ve corrected the issues, you can use Xcode or Application Loader to upload a new binary to iTunes Connect.

Regards,

The App Store team
```

**Comment by janpio** [comment]
Can you provide a link to some documentation how and where this has to be included?
Before this only had to be uploaded in iTunes Connect I think.

**Comment by stalinkay** [comment]
Hi, @janpio. The new icon needs to be included with other icons in the `.appiconset`. Please see [App Icon](https://developer.apple.com/ios/human-interface-guidelines/icons-and-images/app-icon/) and the image below.
<img width="543" alt="screen shot 2017-09-16 at 12 14 26 pm" src="https://user-images.githubusercontent.com/198020/30511431-484b613a-9ad9-11e7-8750-a51f6ec66435.png">

**Comment by tobiasmuecksch** [comment]
Will ionic do that automatically with the `resources` command?

**Comment by stalinkay** [comment]
@tobiasmuecksch If the .appcionset is created by `cordova`, it will require an update from their side but I'm not clued-in as to how it is currently handled. If is handled `ionic` then it will be a matter of creating an additional icon provided the icon artwork is 1024pt or larger.

**Comment by tobiasmuecksch** [comment]
@stalinkay As far as I know; although the command is `ionic cordova resources` the requests are handled by ionic servers, which points me to the assumption, that this is an ionic feature.

**Comment by stalinkay** [comment]
@tobiasmuecksch Yeah, you are right. Just took a pick at the CLI repo. So this should work as before with the added Marketing Icon.

**Comment by jayordway** [comment]
Any resolution on this or when it will be resolved, temporary fix?

**Comment by janpio** [comment]
Temporary fix: Add the file via Xcode manually.

Resolution: I think on the Cordova side this will be handled via https://issues.apache.org/jira/browse/CB-13191 + https://github.com/apache/cordova-ios/pull/337. On the Ionic side where the file has to be generated by `ionic cordoca resources` I am not aware of any work being done (yet), so you would probably still generate and place the file manually.

(But to be honest, this is very confusing what is doing what...)

**Comment by ronaldborla** [comment]
I'm also having this issue. I'll just wait for update from ionic

**Comment by dwieeb** [comment]
Thanks all, the CLI will support generating the marketing icon in the next release! 👍

**Comment by dwieeb** [comment]
Please keep in mind we will need a new release of cordova-ios as well (see https://github.com/apache/cordova-ios/pull/337).

**Comment by stalinkay** [comment]
@dwieeb Quick question, is there a way to specify the quality/compression of the generated icons using `ionic cordova resources`? If not can that be done? I recently generated some icons and splash images with a logo and and it didn't render so well.

**Comment by dwieeb** [comment]
@stalinkay Believe it or not, we use imagemagick for the generated images. You can read about image quality here: https://www.imagemagick.org/script/command-line-options.php#quality. We don't specify the `-quality` argument, which means the quality is inferred from the input image.

If you want, you can open another issue. If you do, would you mind uploading your source image and a generated image that didn't render well?

**Comment by stalinkay** [comment]
@dwieeb Cool.

**Comment by tai-x** [comment]
See: https://stackoverflow.com/a/46480859/1247192 - solution already available on cordova-ios 4.5.1

**Comment by tonydiep** [comment]
That is awesome. Thanks for the hardwork!

**Comment by brassier** [comment]
Are there any tricks/risks to getting cordova-ios 4.5.1 with Ionic?  I'm getting 4.3.1 currently.  I was going to just upgrade it manually but wasn't sure if 4.3.1 is currently the supported platform for Ionic or not?

**Comment by billygerhard** [comment]
The only issue I found so far is the cordova-plugin-console is not compatible with it, so it makes it harder to troubleshoot. I don't know why the cordova-ios update wasn't mentioned in the iOS 11 checklist blog though.

**Comment by eugenedw** [comment]
So, will the "ionic resources" command begin to generate this Marketing Icon? I've run it for the last week or so and am still not getting it.

$ ionic -v
2.2.1

Thanks!

**Comment by janpio** [comment]
You should upgrade to the last Ionic CLI, Cordova CLI and cordova-ios for this to work.

**Comment by GabLeRoux** [comment]
Doesn't seem to work with command line version `2.2.3`. Current latest version of the ionic command line is `3.15.2`, but I think it's kind of a big jump from `npm install ionic@2` to `npm install ionic@3`, Sounds like it'll break things in my project 😱

**Comment by dwieeb** [comment]
@GabLeRoux The changes are documented here: https://github.com/ionic-team/ionic-cli/blob/master/CHANGELOG.md#300-2017-05-09

You can easily revert if the new CLI is still an issue. CLI 2 is aging quite quickly. Please upgrade ASAP.

**Comment by mobilewin0215** [comment]
+1
I also can't upload app to iTunes since the PNG format.

**Comment by tobiasmuecksch** [comment]
@mobilewin0215 Please Update ionic and Cordova like this:

`npm i -g ionic cordova`

**Comment by mobilewin0215** [comment]
I did below steps for update to App Store.
1. sudo npm i -g ionic cordova
2. ionic cordova platform rm ios
3. ionic cordova platform add ios
4. ionic cordova resources
5. cordova build ios

Finally, I did validate in Xcode but I got a message from iTunes.

iTunes Store Operation Failed
Invalid App Store Icon. The App Store Icon in the asset catalog in 'TVC News.app' can't be transparent nor contain an alpha channel.

PS: My project is Ionic1.
Any help? Thanks.
